src/main/java/dk/frankbille/scoreboard/ratings/RatingProvider.java

```diff
@@ -18,13 +18,27 @@
 
 package dk.frankbille.scoreboard.ratings;
 
+import dk.frankbille.scoreboard.ScoreBoardSession;
+import dk.frankbille.scoreboard.domain.RatingCalculatorType;
+import dk.frankbille.scoreboard.domain.User;
 import dk.frankbille.scoreboard.ratings.elo.ELORatingCalculator;
+import dk.frankbille.scoreboard.ratings.trueskill.TrueSkillRatingCalculator;
 
 public class RatingProvider {
 	public static RatingCalculator ratings;
 	public static RatingCalculator getRatings() {
-		if (ratings==null)
-			ratings = new ELORatingCalculator();
+		RatingCalculatorType ratingCalculator = RatingCalculatorType.ELO;
+		if (ScoreBoardSession.get().isAuthenticated()) {
+			User user = ScoreBoardSession.get().getUser();
+			ratingCalculator = user.getRatingCalculator();
+		}
+		if (ratings==null || ratings.getType() != ratingCalculator) {
```

Comment:
Ouch, indeed there is - i'll look into it.
But the problem is much bigger than this class. The system is already using a single shared instance of ``ELORatingCalculator``, which holds state that gets reset every time ``setGames()`` is called, but the consumers expect state to be consistent across their calls to the other exposed methods.